paasta_tools/kubernetes_tools.py

```diff
@@ -1576,6 +1580,11 @@ def get_pod_template_spec(
             for registration in self.get_registrations():
                 labels[f"paasta.yelp.com/registrations/{registration}"] = "true"  # type: ignore
 
+        # Add istio sidecar injection label
+        sidecar_injection_enabled = self.is_istio_sidecar_injection_enabled()
+        if sidecar_injection_enabled:
+            labels["sidecar.istio.io/inject"] = "true"
```

Comment:
we should probably be consistent and do something like
```suggestion
        if self.is_istio_sidecar_injection_enabled():
            labels["sidecar.istio.io/inject"] = "true"
```
 - especially since we're not using that value again later anyway